config/v1/types_apiserver.go

```diff
@@ -173,6 +173,9 @@ type APIServerNamedServingCert struct {
 	ServingCertificate SecretNameReference `json:"servingCertificate"`
 }
 
+// APIServerEncryption is used to encrypt sensitive resources on the cluster.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'KMS' ?  has(self.kms) : !has(self.kms)",message="kms config is required when encryption type is KMS, and forbidden otherwise"
```

Comment:
This has to be the feature gate aware XValidation else it will be added to the default version